### PR TITLE
39 tms carrier wave setup

### LINKED ISSUE #8: Create CouchPhoto model
Validate association - belongs_to :room

attribute: image

### LINKED ISSUE #39: Carrierwave and image hosting

### LINKED ISSUE #74: Host can upload couch photo
As a host with a couch
when I visit the page for my couch
when I click "Add Couch Photos"
and I upload the photo
and click "Update Photo"
then I should be back on the couch page
and I should see that photo

because I want users to be able to see the couch.

### PR DESCRIPTION
**Pull request by tmikeschu**
@Sh1pley @njgheorghita @riverswb @ski-climb ready for review!

* This became a bit of an integrated PR. At a high level, adds ability to upload photos. Specifically:
* Feature test for host adds couch photo
* Refactor routes and controllers to avoid double nesting
* Add photos association for couch (with test)
* Configure carrierwave, fog, and google cloud storage
* Add base content for couch show page (styling needed)
* Fix couch name problem in reservation feature test
* Reformat env vars

All tests pass, and I had to add our google keys to travis (just FYI).

Closes #39 
Closes #74  
Closes #8

### CODE REVIEW

**Review by riverswb — APPROVED** [review]

**Comment by riverswb** [comment]
Looks good to me!

File: Gemfile.lock [hunk 1]
```diff
@@ -92,6 +98,134 @@ GEM
...
+      fog-core
+      rbvmomi (~> 1.9)
+    fog-xenserver (0.2.3)
+      fog-core
+      fog-xml
+    fog-xml (0.1.2)
+      fog-core
+      nokogiri (~> 1.5, >= 1.5.11)
+    formatador (0.2.5)
     globalid (0.3.7)
```
Comment (njgheorghita) [inline Gemfile.lock:229]: jeez fog

File: spec/features/reservations/traveler_can_make_a_reservation_spec.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@
   describe "As a registered user" do
     let!(:traveler) { create(:user) }
     let!(:couch_1)  { create(:couch, city: "Another City") }
-    let!(:couch_2)  { create(:couch, city: "Mike's Hometown", name: "CRAZY NAME") }
+    let!(:couch_2)  { create(:couch, city: "Mike's Hometown", name: "NEVER GONNA REPEAT") }
```
Comment (njgheorghita) [inline spec/features/reservations/traveler_can_make_a_reservation_spec.rb:18]: Cool, thanks for the fix!

File: app/controllers/couches/photos_controller.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+    @photo = CouchPhoto.new
+  end
+
+  def create
+
+    couch = Couch.find(params["couch_id"])
+    @photo = couch.photos.new(photo_params)
+    if @photo.save!
+      redirect_to user_couch_path(current_user, couch)
+    end
```
Comment (njgheorghita) [inline app/controllers/couches/photos_controller.rb]: else flash[:danger] & redirect_to new_couch_path? though we can easily throw this in down the road

File: app/views/couches/photos/new.html.haml [hunk 4]
```diff
@@ -0,0 +1,15 @@
+= form_for @photo do |f|
+  = f.label :title
+  = f.text_field :title
+
+  = f.label :caption
+  = f.text_field :caption
+
```
Comment (njgheorghita) [inline app/views/couches/photos/new.html.haml:7]: do we want to insert a caption that it only accepts filetypes jpg, jpeg, gif, png?

File: config/initializers/carrierwave.rb [hunk 5]
```diff
@@ -0,0 +1,9 @@
+CarrierWave.configure do |config|
+  config.fog_credentials = {
+    provider: 'Google',
+    google_storage_access_key_id: ENV['google_storage_access_key_id'],
+    google_storage_secret_access_key: ENV['google_storage_secret_access_key']
```
Comment (njgheorghita) [inline config/initializers/carrierwave.rb:5]: way to hide dem keys!

File: spec/features/photos/photo_upload_spec.rb [hunk 6]
```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.feature "Photo uploads" do
+  context "Host" do
+    before do
+        Fog.mock!
```
Comment (njgheorghita) [inline spec/features/photos/photo_upload_spec.rb:6]: Would love a talkthrough on this test whenever possible

**Comment by njgheorghita** [comment]
Awesome @tmikeschu. Now we can definitively prove that we have the best couches. Nobody has couches like ours. Airbnb couches. SAD!!

File: app/controllers/couches/photos_controller.rb (hunk 3 above)
Comment (tmikeschu) [reply app/controllers/couches/photos_controller.rb]: @njgheorghita added!

File: spec/features/photos/photo_upload_spec.rb (hunk 6 above)
Comment (tmikeschu) [reply spec/features/photos/photo_upload_spec.rb:6]: 👍

**Comment by tmikeschu** [comment]
Functionality also confirmed in the app:

screen shot 2017-02-12 at 10 54 23 am
screen shot 2017-02-12 at 10 54 40 am

**Pull request merged**